flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -508,7 +508,9 @@ public void killCluster(ApplicationId applicationId) throws FlinkException {
 			throw new YarnDeploymentException("Could not retrieve information about free cluster resources.", e);
 		}
 
-		final int yarnMinAllocationMB = yarnConfiguration.getInt(YarnConfiguration.RM_SCHEDULER_MINIMUM_ALLOCATION_MB, 0);
+		final int yarnMinAllocationMB = yarnConfiguration.getInt(
```

Comment:
I put in `validateClusterResources` because I thought it's a part of the validation, but I'm also okay with your way.

Reply:
I think it would be good to always try to throw an exception with the most descriptive type if possible. IMO, this should have higher priority than keeping the codes concise.

In this case, I would suggest to throw either a `YarnDeploymentException` or a `IllegalConfigurationException`, rather than `IllegalStateException` that `Preconditions.checkArgument` throws.